Marlin/Marlin_main.cpp

```diff
@@ -4176,83 +4203,93 @@ inline void gcode_M105() {
+    serialprintPGM(PSTR(MSG_T_CARTRIDGE_REMOVED_HEATING));
+    SERIAL_EOL;
+    SERIAL_ECHOLN("// action:pause");
+  }
+  else
+  {
+    if (setTargetedHotend(109)) return;
+    if (marlin_debug_flags & DEBUG_DRYRUN) return;
 
-  LCD_MESSAGEPGM(MSG_HEATING);
+    LCD_MESSAGEPGM(MSG_HEATING);
 
-  no_wait_for_cooling = code_seen('S');
-  if (no_wait_for_cooling || code_seen('R')) {
-    float temp = code_value();
-    setTargetHotend(temp, target_extruder);
-    #if ENABLED(DUAL_X_CARRIAGE)
-      if (dual_x_carriage_mode == DXC_DUPLICATION_MODE && target_extruder == 0)
-        setTargetHotend1(temp == 0.0 ? 0.0 : temp + duplicate_extruder_temp_offset);
-    #endif
-  }
+    no_wait_for_cooling = code_seen('S');
+    if (no_wait_for_cooling || code_seen('R')) {
+      float temp = code_value();
+      setTargetHotend(temp, target_extruder);
+      #if ENABLED(DUAL_X_CARRIAGE)
+        if (dual_x_carriage_mode == DXC_DUPLICATION_MODE && target_extruder == 0)
+          setTargetHotend1(temp == 0.0 ? 0.0 : temp + duplicate_extruder_temp_offset);
+      #endif
+    }
 
-  #if ENABLED(AUTOTEMP)
-    autotemp_enabled = code_seen('F');
-    if (autotemp_enabled) autotemp_factor = code_value();
-    if (code_seen('S')) autotemp_min = code_value();
-    if (code_seen('B')) autotemp_max = code_value();
-  #endif
+    #if ENABLED(AUTOTEMP)
+      autotemp_enabled = code_seen('F');
+      if (autotemp_enabled) autotemp_factor = code_value();
+      if (code_seen('S')) autotemp_min = code_value();
+      if (code_seen('B')) autotemp_max = code_value();
+    #endif
 
-  millis_t temp_ms = millis();
+    millis_t temp_ms = millis();
 
-  /* See if we are heating up or cooling down */
-  target_direction = isHeatingHotend(target_extruder); // true if heating, false if cooling
+    /* See if we are heating up or cooling down */
+    target_direction = isHeatingHotend(target_extruder); // true if heating, false if cooling
 
-  cancel_heatup = false;
+    cancel_heatup = false;
 
-  #ifdef TEMP_RESIDENCY_TIME
-    long residency_start_ms = -1;
-    /* continue to loop until we have reached the target temp
-      _and_ until TEMP_RESIDENCY_TIME hasn't passed since we reached it */
-    while((!cancel_heatup)&&((residency_start_ms == -1) ||
-          (residency_start_ms >= 0 && (((unsigned int) (millis() - residency_start_ms)) < (TEMP_RESIDENCY_TIME * 1000UL)))) )
-  #else
-    while ( target_direction ? (isHeatingHotend(target_extruder)) : (isCoolingHotend(target_extruder)&&(no_wait_for_cooling==false)) )
-  #endif //TEMP_RESIDENCY_TIME
+    #ifdef TEMP_RESIDENCY_TIME
+      long residency_start_ms = -1;
+      /* continue to loop until we have reached the target temp
+        _and_ until TEMP_RESIDENCY_TIME hasn't passed since we reached it */
+      while(((!cancel_heatup)&&((residency_start_ms == -1) ||
+            (residency_start_ms >= 0 && (((unsigned int) (millis() - residency_start_ms)) < (TEMP_RESIDENCY_TIME * 1000UL))))) && !CartridgeRemovedFFF())
```

Comment:
I know Marlin is a disaster tostart with, but please format this line if you are modifying this code at all